[Wang Jingting] iP

jingting1412 opened the pull request:
# Dukey 💆🏻
> "Aim for the moon. If you miss, you may hit a star" – William Clement Stone ([source](https://www.alphatrad.com/news/famous-phrases-in-english))

Dukey is here to help you organise tasks! He is:

- text-based  💬
- easy to learn ⚡️
- FAST SUPER FAST to use 🔥

All you need to do is,

1. download it from [this jar file](https://github.com/jingting1412/ip/releases).
1. open it on your own computer
1. let it manage your tasks for you 😉

## And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) throws DukeException {
        Duke dukey = new Duke();
        dukey.run("duke.txt");
}
```

***Here are some useful commands***
- `todo`
-  `deadline` `/by` 
-  `event` `/from` `/to`
-  `delete`
-  `find`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
changed .gitignore file

Commit:
implementing a skeleton version

Commit:
add indentations and lines

Commit:
Echo

Commit:
Add, List

Commit:
Add Task class

Commit:
Mark as Done

Commit:
ToDos, Events, Deadlines

Commit:
Automated Text UI Testing

Commit:
Handle Errors

Commit:
Delete

Commit:
Add function to save to file

Commit:
Load data when chatbot starts

Commit:
Save tasks and load data

Commit:
Edit duke file

Commit:
Save to file, load from file

Commit:
Fix indentation

Commit:
Merge branch 'branch-Level-7'

Commit:
Add duke file

Commit:
Store deadline dates as java.time.LocalDate

Commit:
Format date display

Commit:
Merge branch 'branch-Level-8'

Commit:
Add more OOP

Commit:
Organize into Packages

Commit:
Add gradle support

Commit:
Add gradle support

Commit:
Set up gradle

Commit:
Add DukeException file

Commit:
Add test cases

Commit:
Add JavaDocs

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Follow coding standard

Commit:
Format code to follow coding standard

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Find task by searching for a keyword

Commit:
Improve find method

Commit:
Merge branch 'branch-Level-9'

Commit:
Add checkstyle

Commit:
Merge branch 'branch-A-CheckStyle'

bhnuka commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,112 @@
+    public void parseAndRespond() throws DukeException {
+        String[] splitted = userCommand.split(" ", 2);
+        switch (splitted[0]) {
+        case "list":
```
Used the correct coding standard formatting! Nice 😄

bhnuka commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,112 @@
+                throw new DukeException("Please provide a description for this deadline! (⋟﹏⋞)");
+            } else {
+                String[] deadTask = splitted[1].split("/by");
+                if (deadTask.length == 1) {
```
Could be considered deep nesting with affects readability. Maybe consider employing a command class?

jingting1412 replied:
Good idea, I'll try to create a command class for all the commands and their responses.

bhnuka commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,100 @@
+    /**
+     * Constructs new <code>Storage</code> object.
+     * @param filepath the filepath to the file that stores existing tasks.
+     *                 New file created if the file specified does not exist at the start.
```
Awkward space here... Unsure why

bhnuka commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,100 @@
+    /**
+     * Loads data from the hard disk when the chatbot starts up.
+     * Saves any pre-existing tasks in the file to the task list.
+     * @throws DukeException when data from the file cannot be converted to a <code>Task</code> object.
```
There should be an empty line between the summary and parameters/return/thrown

bhnuka commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,120 @@
+    /**
+     * Deletes a specific task from the task list.
+     * @param target the index of the task in the task list that should be deleted.
+     * @throws DukeException throws exception when the index of the task given is larger than the size of the task list.
```
Just on the brink of the hard limit for line character count (120). Would be best to separate into two lines to improve readability.

jingting1412 replied:
Will do, thanks for the suggestion

bhnuka commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,120 @@
+        List<Task> result = toDo.stream()
+                .filter(task -> task.checkTerm(searchTerm))
+                .collect(Collectors.toList());
+        System.out.println("        ____________________________________________________________"
```
The line should be under UI, not TaskList.

bhnuka reviewed (commented):
Overall, very intelligible code. Minor nits here and there but they're quick fixes.  Keep it up!

Commit:
Implement GUI

Commit:
Implement basic GUI

Commit:
Debug GUI

Commit:
Set Launcher to Main

Commit:
Improve UI of Duke

Commit:
Improve UI

Commit:
Merge branch 'branch-Level-10'

Commit:
Add assertions

jingting1412 committed:
Merge pull request #1 from jingting1412/branch-A-Assertions

Add assertions

Commit:
Improve code quality

jingting1412 committed:
Merge pull request #2 from jingting1412/branch-A-CodeQuality

Improve code quality

Commit:
Check for duplicates

TaskList add command: check for duplicate tasks

The add method does not check for duplicates, allowing users to add the
same task multiple times.

Adding the ability to recognise and detect duplicates allows for a
cleaner task list.

Let's add the ability to detect duplicates by checking if there is
already a task with the same description in the task list. If there is
such a task present, tell the user that there is already an existing
task with the same description in the task list.

This prevents users from adding tasks with the same description over and
over again in the task list.

Commit:
Revert "Check for duplicates"

This reverts commit b326993997ed1c39920dcec9884de9b42d4a00b4.

Commit:
Check for duplicates

TaskList add command: check for duplicate tasks

The add method does not check for duplicates, allowing users to add the
same task multiple times.

Adding the ability to recognise and detect duplicates allows for a
cleaner task list.

Let's add the ability to detect duplicates by checking if there is
already a task with the same description in the task list. If there is
such a task present, tell the user that there is already an existing
task with the same description in the task list.

This prevents users from adding tasks with the same description over and
over again in the task list.

Commit:
Integrate isDuplicate with add method

add method: does not check if the task to be added is a duplicate

The add method does not check for duplicates, allowing users to add the
same task multiple times.

Let's integrate the isDuplicate method with the add method to check if
there is already a task in the task list matching the description of the
task given to the add method.

Commit:
Merge branch 'branch-BCD-Extension'

Branch master: No ability to recognise and deal with duplicate items

This allows the user to add a task multiple times.

Let's merge the branch `branch-BCD-Extension` which implemented the
DetectDuplicates function to the branch `master`. If there is
such a task present, the chatbot will inform the user that there is
already an existing task with the same description in the task list.

This prevents users from adding tasks with the same description over and
over again in the task list, creating a cleaner and more readable task
list.

Commit:
Add folders

Commit:
Improve coding standard

Commit:
Improve GUI

Commit:
Add UG

Commit:
Fix emoticon compatibility issues

Commit:
Fix dependecy issues

jingting1412 committed:
Update README.md

jingting1412 committed:
Update README.md

jingting1412 committed:
Update README.md

jingting1412 committed:
Update README.md

jingting1412 committed:
Update README.md
